lib/commands/command-info.bash

```diff
@@ -4,9 +4,13 @@
 
 info_command() {
   printf "%s:\n%s\n\n" "OS" "$(uname -a)"
-  printf "%s:\n%s\n\n" "SHELL" "$($SHELL --version)"
+  printf "%s:\n%s\n\n" "SHELL" "Bash $BASH_VERSION"
```

Comment:
I use a non-Bash shell as a login shell and the existing behaviour works as expected. I understand we might want to know the BASH_VERSION as well, but knowing the login Shell is just as useful a debugging mechanism.

Perhaps we can better handle failures with `$SHELL` and then also add `$BASH_VERSION` separately?

Reply:
> I use a non-Bash shell as a login shell and the existing behaviour works as expected.

Ohh, oops what I said is only true for interactive logins I believe.

> I would argue when people provide this information we want the more verbose output. Bash itself being more verbose than other shells is a Bash-ism.

What part of the information is useful? All lines except the first are license boilerplate and the stuff in the target triple is covered by `uname -a`

> Perhaps we can better handle failures with $SHELL and then also add $BASH_VERSION separately?

If you still don't prefer `$BASH_VERSION`, then I'll just quote `"$SHELL"` and we can leave it at that. I also opened an issue in Shellcheck https://github.com/koalaman/shellcheck/issues/2724 so hopefully this kind of stuff gets caught in the future

Reply:
>Ohh, oops what I said is only true for interactive logins I believe.

Right, well `asdf info` was created for ease of sharing data for Issues/PRs in the repo. It is expected to be used in an interactive shell. Uses for scripting or non-interactive environments will eventually fall under a particular global flag (tentatively `--porcelain`) and so is not a goal of the current implementation.

>What part of the information is useful? All lines except the first are license boilerplate and the stuff in the target triple is covered by `uname -a`

In my Nushell setup, `$($SHELL --version)` gives me just my Nushell version number, nothing else. `$BASH_VERSION` while also useful information, won't share my current Shell information in `asdf info`. That information about my Nushell version is lost. This is useful information for us to debug with during Issues which `asdf info` was created for ease of sharing with.

As I said, `$BASH_VERSION` is useful as well, so seeing it as an addition to `$($SHELL --version)` is welcome.

Removal of `$($SHELL --version)` results in a loss of information.

So **adding** `$BASH_VERSION` is welcome :+1: Better error handling of `$($SHELL --version)` is also welcome, but it is preferred to keep it. The `--version` verbosity is useful with non-Bash shells, like Nushell.

I hope that clarifies things.

Reply:
I see - using `$SHELL` is important and the fact that it potentially invokes a different shell didn't register with me. I'll leave it in and add `$BASH_VERSION`